spec/system/consumer/split_checkout_spec.rb

```diff
@@ -207,10 +207,109 @@
         end
       end
 
-      describe "filling out delivery details" do
+      describe "filling out order details" do
         before do
           fill_out_details
           fill_out_billing_address
+          order.update(user_id: user.id)
+        end
+
+        context "billing address" do
+          describe "checking the default address box" do
+            before do
+              check "order_save_bill_address"
+              choose free_shipping.name
+            end
+
+            it "before saving has no default address" do
+              expect(user.bill_address).to be_nil
+              expect(order.bill_address).to be_nil
+            end
+
+            context "as a first time customer" do
+              before do
+                proceed_to_payment
+              end
+              it "creates a new default bill address" do
+                expect(order.reload.bill_address.address1).to eq "Rue de la Vie, 77"
+                expect(order.customer.bill_address.address1).to eq "Rue de la Vie, 77"
+                expect(user.reload.bill_address.address1).to eq "Rue de la Vie, 77"
+              end
+            end
+          end
+
+          describe "unchecking the default address box" do
+            before do
+              uncheck "order_save_bill_address"
+              choose free_shipping.name
+            end
+
+            context "as an existing customer" do
+              let(:existing_address) { create(:address) }
+
+              before do
+                user.bill_address = existing_address
+                user.save
+                proceed_to_payment
```

Comment:
These are quite good examples for the `change` helper. You could do this in the actual test:

```suggestion
                expect {
                  proceed_to_payment
                }.to_not change {
                  user.reload.bill_address
                }
```

Reply:
To my understanding, I think the `proceed_to_payment` step is needed so that the DB is changed. The `save!` is meant here as a starting point for the test. So, if we don't proceed to the payment, there should be no change in the DB. Right?

Reply:
Ahh, I get now the direction of this comment now, together with your previous one. If we use the change helper then we don't need a new address :+1: 

I think I'll just clarify the context lines a bit. These changes also make clear that we are not really testing `first time or existing customers`. We're testing the checkbox behavior, so I think this is now clear after this last commit 6392630c0.